src/coreclr/System.Private.CoreLib/src/System/Reflection/TypeNameParser.CoreCLR.cs

```diff
@@ -0,0 +1,238 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.Loader;
+using System.Text;
+using System.Threading;
+
+namespace System.Reflection
+{
+    internal unsafe ref partial struct TypeNameParser
+    {
+        private ReadOnlySpan<char> _input;
+        private int _index;
+        private int _errorIndex; // Position for error reporting
+
+        private Func<AssemblyName, Assembly?>? _assemblyResolver;
+        private Func<Assembly?, string, bool, Type?>? _typeResolver;
+        private bool _throwOnError;
+        private bool _ignoreCase;
```

Comment:
Why not to move shared fields to shared location too? Also, some of them could be made readonly.

Reply:
Good idea. I had to also add `LayoutKind.Auto` to stop Roslyn from complaining about unspecified field order.